docs/spec_draft.md

```diff
@@ -111,6 +111,15 @@ implementation-defined, as long as ops are evaluated before their uses. Possible
 execution orders of the above example program are `%0` → `%1` → `%2` → `return`
 or `%1` → `%0` → `%2` → `return`.
 
+### Replica and Partition of a program
+
+Multiple instances of a compiled StableHLO program could be running to exploit
+data parallelism. Each instance is called a replica identified uniquely using
+replica id (`rid`). In addition, several different StableHLO programs can be
+executed together to exploit "Multiple Program Multiple Data" (MPMD) style
+parallelism and these are called partitions, each of which is identified uniquely
+using a partition id (`pid`).
+
```

Comment:
What's the rationale behind calling them "partition"? Are you considering the total execution as a mesh/graph of dependent functions and each "executing program" is a partition of that mesh/graph?
Why is "program" not enough?

Reply:
Yeah, I've got the same question when reviewing this PR, and I don't think we need to categorize these programs into replicas and partitions. At least, not for speccing collective ops.